CleanWeather/Scenes/CityList/CityListWorker.swift

```diff
@@ -30,16 +32,36 @@ extension CityListWorkerImpl: CityListWorker {
         repository.fetchFavouriteCities(completion: completion)
     }
 
-    func fetchCityWeather(completion: FetchWeatherCompletion?) {
-        networking.fetchCurrentWeatherForAllCities { [weak self] result in
-            
-            guard let self = self, case .success(let city) = result else {
-                completion?(result)
-                return
+    func fetchCitiesWeather(cities: [City], completion: FetchWeatherCompletion?) {
+
+        var counter = 0
+        var citiesWeather = [CityWeather]()
+
+        cities.forEach { city in
+            let coordinates = Coordinates(lat: city.latitude, lon: city.longitude)
+            networking.fetchCurrentWeatherForCity(coordinates: coordinates.stringValue) { result in
+                switch result {
+                case .success(let apiResponse):
+                    guard let cityTemp = apiResponse.currently.temperature else {
+                        completion?(.failure(MissingAPIData()))
+                        return
+                    }
+                    let cityWeather = CityWeather(id: city.id,
+                                                  city: city.name,
+                                                  latitude: city.latitude,
+                                                  longitude: city.longitude,
+                                                  temperature: cityTemp,
+                                                  icon: apiResponse.currently.icon ?? "")
+                    citiesWeather.append(cityWeather)
+                    counter += 1
+                    if counter == cities.count {
+                        let sortedCities = self.sortCity(city: citiesWeather)
+                        completion?(.success(sortedCities))
+                    }
+                case .failure(let error):
+                    completion?(.failure(error))
```

Comment:
this can be called multiple times
make your own local completion block where u count number of responses and handle errors